src/sql/src/rbac.rs

```diff
@@ -259,23 +188,28 @@ pub fn check_plan(
 ) -> Result<(), UnauthorizedError> {
     rbac_preamble!(catalog, role_metadata, session_vars);
 
-    check_item_usage(
+    // Obtain all roles that the current session is a member of.
+    let role_membership = catalog.collect_role_membership(&role_metadata.current_role);
+
+    let rbac_requirements = generate_rbac_requirements(
```

Comment:
What do you think about moving all of the "checking logic" into a method on `RbacRequirements`? So this could would look something like:

```
let rbac_requirements = generate_rbac_requirements(...);

rbac_requirements.validate(...)?;
```

It's "half dozen of one, six of the other" so I'm good either way, it's just a thought I had while looking at this

Reply:
Yeah, I think it's a good idea, I just moved the logic.